tests/api/test_registry.py

```diff
@@ -99,7 +102,9 @@ def test_registry():
         "kubernetes-app": KubernetesApp,
         "pipeline-component": PipelineComponent,
         "producer-app": ProducerApp,
+        "producer-app-v3": ProducerAppV3,
         "streams-app": StreamsApp,
+        "streams-app-v3": StreamsAppV3,
```

Comment:
I had to suffix the streamApp and producerApp classes with v3 otherwise the registry would overwrite the streams bootstrap v2 classes

Reply:
if streams-bootstrap v3 is going to become the new default, I would prefer keeping `streams-app` for v3 and adding the -v2 suffix to the old one (breaking change)

Reply:
I agree with you, but I would instead do it in a follow-up PR. We can create a new branch, v8, and do the changes step-by-step to keep the PRs small.